db/migrate/20220216132652_populate_upstream_authentication_token.rb

```diff
@@ -0,0 +1,21 @@
+class PopulateUpstreamAuthenticationToken < ActiveRecord::Migration[6.0]
+  def up
+    return unless ActiveRecord::Base.connection.column_exists?('katello_root_repositories', 'upstream_authentication_token')
```

Comment:
True, if the Katello version is upgraded after this is run, then the error will persist. We could add a rake task for the migration that can be run on demand. At the moment, this will only work if someone has an old system, upgrades to a new Katello version and then upgrades SCC manager. 

We could also make the migration as `not run` in case the Katello version is too old. No idea how that can be done, though.

Reply:
It is a tricky problem, if we do this as a rake-task, we must make sure that it runs, when we (or more precisely katello) need it to.

I hesitate to propose doing this as a seed-task, which would make sure it is tried to run pretty much after every update.
On the other hand it seems to be overkill to always try to run this migration just to hit the one occasion where it is really needed :thinking:

Reply:
None that I know of.

Reply:
Then I propose to implement a rake task and document it well. Otherwise, we need to check when Katello is updated and to which version.